src/query/ast/src/ast/expr.rs

```diff
@@ -233,9 +233,11 @@ pub enum Literal {
 #[derive(Debug, Clone, PartialEq)]
 pub enum MapAccessor<'a> {
     /// `[0][1]`
-    Bracket { key: Literal },
+    Bracket { key: Box<Expr<'a>> },
     /// `.a.b`
     Period { key: Identifier<'a> },
+    /// `.1`
+    PeriodNumber { key: u64 },
```

Comment:
this is only used for tuple so far... because it has different sematic to Braket.

.<number> must be a const, while [<expr>] can be any arbitrary expression including column ref.

Reply:
ok